featurebyte/worker/task_executor.py

```diff
@@ -193,7 +193,18 @@ async def execute(self) -> Any:
                 )
 
             # Send final progress to indicate task is completed
-            await self.task_progress_updater.update_progress(percent=100)
+            task = await self.task_manager.get_task(task_id=str(self.task_id))
+            skip_last_progress_update = False
+            if (
+                task is not None
+                and task.progress is not None
+                and task.progress.get("percent") == 100
+            ):
+                # skip the last progress update if the progress is already 100%
+                skip_last_progress_update = True
+
+            if not skip_last_progress_update:
+                await self.task_progress_updater.update_progress(percent=100)
```

Comment:
Current sub-tasks rely on progress's metadata field to pass the information back to parent task. If the task's progress get overridden, the metadata value will be lost.

Reply:
Would it make sense to update the `update_progress` logic to retain existing message if the `message` parameter is not provided.

Reply:
Do you mean revise this fix to keep the existing metadata & message & only update percentage?

Reply:
Would it address the issue that you are trying to fix? Does it make sense for a more systematic fix to similar problems? If not we can keep the current approach

Reply:
The task will automatically update the percentage to 100% and override message & task metadata. If we want to change `update_progress` method, we need to do something like keep the last value if it is not provided. Just not sure whether there is any case where we want to set the last update message/metadata to be `None`.

Reply:
Indeed. That will require an additional parameter. Let's keep the current approach